graph/src/components/store.rs

```diff
@@ -234,6 +270,27 @@ pub struct StoreEvent {
     pub changes: HashSet<EntityChange>,
 }
 
+impl From<Vec<EntityOperation>> for StoreEvent {
+    fn from(operations: Vec<EntityOperation>) -> Self {
+        let changes: Vec<_> = operations.into_iter().map(|op| op.into()).collect();
+        StoreEvent::new(changes)
+    }
+}
+
+impl From<Vec<MetadataOperation>> for StoreEvent {
+    fn from(operations: Vec<MetadataOperation>) -> Self {
+        let changes: Vec<_> = operations.into_iter().filter_map(|op| op.into()).collect();
+        StoreEvent::new(changes)
+    }
+}
+
+impl From<&Vec<EntityModification>> for StoreEvent {
```

Comment:
Avoiding the clone is the point of passing the reference here since we do this _a lot_ and we'd really only clone for notational convenience.

Reply:
I checked, we use this impl only one time.

Reply:
If this were cheap to clone it would be merely an idiomatic thing, but since this is a potentially large vector so potentially expensive to clone, surfacing the clone to the caller makes it easier to reason about performance if necessary.

Reply:
I don't understand what you are getting at. My objection was that the caller would have to clone solely so that we can write `Vec<_>` instead of `&Vec<_>` here, and that we should avoid the clone.

Reply:
On a more careful look, the original clone obfuscation is happening in `impl From<&EntityModification> for EntityChange`. We need to remove the `&` from that impl, and then remove it from this impl, and then it will be clear to us and to caller that we are in fact cloning here.

Reply:
Addressed with `FromIterator` in commit fbbf55de